### ISSUE TITLE
How 'index_to_example_function' works with 'log_confusion_matrix' in PyTorch

### ISSUE THREAD

**Issue by kuri54** [comet-ml/comet-examples#35]
torch==1.8.1+cu111
torchaudio==0.8.1
torchvision==0.9.1+cu111
comet-ml==3.9.0

I find comet_ml very easy to use and have been studying and using it since the tutorial.
However, the problem is that the `index_to_example_function` of `log_confusion_matrix` does not work as expected.

code:
```
def valid_index_to_example(index):
    tmp, _ = image_datasets['valid'][index].
    img = tmp.numpy()[0].
    data = experiment.log_image(img, name="valid_%d.png" % index)

    if data is None:
        return None

    return {"sample": str(index), "assetId": data["imageId"]}

・
・Omission of code
・
・

experiment.log_confusion_matrix(labels, preds,
                                labels=class_names, 
                                title='Confusion Matrix, Epoch #{}'.format(epoch),
                                file_name='confusion-matrix-{}.json'.format(epoch),                                                                 
                                index_to_example_function=valid_index_to_example)

```

However, if you set `shuffle=True` for dataloader, it will refer to the wrong image.
Of course, if you set `shuffle=False`, it will refer to the correct image, but the training results will not be good, so you want to set `shuffle=True`.

comet_ml's PyTorchs tutorial describes how to callback Keras and TensorFlow, but not PyTorch.
Is there any way to solve this?

I apologize if my English does not convey my intentions.

**Comment by dsblank** [comment]
That was very clear, and you are correct: pytorch does not give a standard way to give you a global index that refers to the same inputs (and targets) regardless of randomization.

However, it is not impossible. If you run `comet init` at the command-line and answer a few questions (making sure to select pytorch and confusion matrix) it will write some sample code for you. The secret is to alter the dataset class to return the index position. Once you do that, then you can see how the included index_to_example function works properly.

There are other ways to, perhaps, but we find this way the easiest.

Let us know if you have any questions or problem with this.

**Comment by kuri54** [comment]
Thanks !

> That was very clear, and you are correct: pytorch does not give a standard way to give you a global index that refers to the same inputs (and targets) regardless of randomization.   

I also think that this was a factor in this case and it did not work well.

It's a holiday in Japan so I'll try to access the analysis server later in the week and try `comet init`.
If I have any questions after using it, I will ask again.

I think comet_ml is a great tool.  I've already switched from TensorBoard, and once I get this sorted out, I can move over to comet_ml completely!
